### DISCUSSION TITLE
Is there a tool that can save the code generated by the agent as a local file

### DISCUSSION THREAD

**Discussion by Frog863** [crewAIInc/crewAI#1093] [Q&A]

**Comment by theCyberTech** [comment]
You can create your own tools, please see the docs - https://docs.crewai.com/core-concepts/Tools/

Also there are attributes on the crew,  tasks and agents you can set to save to file see here

Tasks - https://docs.crewai.com/core-concepts/Tasks/#task-attributes
crew - https://docs.crewai.com/core-concepts/Crews/#crew-attributes

**Reply by Frog863** [reply]
The docs is too sketchy.The existing tools in the file are mainly for reading or searching information and feeding it to the LLM, so I'm still not clear on how to save some of the LLM's outputs locally.
